### DISCUSSION TITLE
How to add a pydantic class as a dependency for a GET route?

### DISCUSSION THREAD

**Discussion by pbizimis** [fastapi/fastapi#8738] [Questions] [answered]
I want to have a GET route that returns the schema of a pydantic class. This way the frontend can call the get route and can build a form that then posts to the post route. A simple version could be:

```
class GenerationOptions(BaseModel):
    model: str
    truncation: float

@router.post("/generate")
def generate_image(model: GenerationOptions):
    return something

@router.get("/generate")
def generate_image(model: dict = Depends(GenerationOptions.schema))): # doesn't work
    return {"form_fields": model}
```

However, this doesn't work. The only way I can achieve what I want is with another function:

```
class GenerationOptions(BaseModel):
    model: str
    truncation: float

@router.post("/generate")
def generate_image(model: GenerationOptions):
    return something

def getSomething():
    return GenerationOptions.schema()

@router.get("/generate")
def generate_image(model: dict = Depends(getSomething)):
    return {"form_fields": model}
```
This works but I think that there might be another/better solution. Any ideas? Thanks!

**Comment by rotten** [comment]
Can you get it from the swagger endpoint instead?

**Comment by pbizimis** [comment]
Wouldn't that be unnecessary? Because I would need to request the complete openapi spec but I only need the parameters of one route.

**Comment by Mause** [comment, marked as answer]
Why are you using Depends to retrieve the schema for the model? Why not just retrieve it directly?

**Comment by pbizimis** [comment]
Mostly because the dependency injection makes it easier to test the route.

**Comment by pbizimis** [comment]
My approach now is this:
```py
class GenerationOptions(BaseModel):
    model: str
    truncation: float

    @classmethod
    def get_schema(cls):
        return cls.schema()

@router.post("/generate")
def generate_image(model: GenerationOptions):
    return model

@router.get("/generate")
def generate_image(model: dict = Depends(GenerationOptions.get_schema)):
    return {"form_fields": model}
```

I am not sure if this is the best way to do this but this works perfectly for my case.
